src/html/link.ml

```diff
@@ -48,46 +67,55 @@ let rec drop_shared_prefix l1 l2 =
+      match resolve with
+      | Base xref_base_uri ->
+          let page = xref_base_uri ^ String.concat "/" target_loc in
+          add_anchor page
+      | Current path -> (
+          let current_loc =
+            let dir, file = Path.get_dir_and_file ~config path in
+            dir @ [ file ]
+          in
 
-      let current_from_common_ancestor, target_from_common_ancestor =
-        drop_shared_prefix current_loc target_loc
-      in
+          let current_from_common_ancestor, target_from_common_ancestor =
+            drop_shared_prefix current_loc target_loc
+          in
 
-      let relative_target =
-        match current_from_common_ancestor with
-        | [] ->
-            (* We're already on the right page *)
-            (* If we're already on the right page, the target from our common
-                ancestor can't be anything other than the empty list *)
-            assert (target_from_common_ancestor = []);
-            []
-        | [ _ ] ->
-            (* We're already in the right dir *)
-            target_from_common_ancestor
-        | l ->
-            (* We need to go up some dirs *)
-            List.map (fun _ -> "..") (List.tl l) @ target_from_common_ancestor
-      in
-      let remove_index_html l =
-        match List.rev l with
-        | "index.html" :: rest -> List.rev ("" :: rest)
-        | _ -> l
-      in
-      let relative_target =
-        if Config.semantic_uris config then remove_index_html relative_target
-        else relative_target
-      in
-      match (relative_target, anchor) with
-      | [], "" -> "#"
-      | page, "" -> String.concat "/" page
-      | page, anchor -> String.concat "/" page ^ "#" ^ anchor)
+          let relative_target =
+            match current_from_common_ancestor with
+            | [] ->
+                (* We're already on the right page *)
+                (* If we're already on the right page, the target from our common
+                    ancestor can't be anything other than the empty list *)
+                assert (target_from_common_ancestor = []);
+                []
+            | [ _ ] ->
+                (* We're already in the right dir *)
+                target_from_common_ancestor
+            | l ->
+                (* We need to go up some dirs *)
+                List.map (fun _ -> "..") (List.tl l)
+                @ target_from_common_ancestor
+          in
+          let remove_index_html l =
+            match List.rev l with
+            | "index.html" :: rest -> List.rev ("" :: rest)
+            | _ -> l
+          in
+          let relative_target =
+            if Config.semantic_uris config then
+              remove_index_html relative_target
+            else relative_target
+          in
+          match (relative_target, anchor) with
+          | [], "" -> "#"
+          | page, "" -> String.concat "/" page
+          | page, anchor -> String.concat "/" page ^ "#" ^ anchor))
```

Comment:
Feel free to ignore: We can save one line by reusing `add_anchor` :)
```diff
-           | page, "" -> String.concat "/" page
-           | page, anchor -> String.concat "/" page ^ "#" ^ anchor))
+           | page, _ -> add_anchor @@ String.concat "/" page))
```

Reply:
Good idea!